### PR TITLE
Support keep_first_word option

### PR DESCRIPTION
**Pull request by erasche**
Wildly useful tool, however keeping the whole description isn't great. The substring option is nice however only works for fasta files with identical length IDs which is ... basically 0% of the ones we encounter.

This exposes the existing option in code to just use the first "word", (so the fasta ID but not the fasta description).

### CODE REVIEW

File: tools/fasta_compute_length/fasta_compute_length.xml [hunk 1]
```diff
@@ -1,9 +1,13 @@
 <tool id="fasta_compute_length" name="Compute sequence length" version="1.0.0">
 	<description></description>
-	<command interpreter="python">fasta_compute_length.py $input $output $keep_first</command>
+	<command interpreter="python">fasta_compute_length.py $input $output $keep_first $keep_first_word</command>
 	<inputs>
 		<param name="input" type="data" format="fasta" label="Compute length for these sequences"/>
 		<param name="keep_first" type="integer" value="0" label="How many title characters to keep?" help="'0' = keep the whole thing"/>
+        <param name="keep_first_word" type="boolean" truevalue="id_only" falsevalue="id_and_desc"
```
Comment (nsoranzo) [inline tools/fasta_compute_length/fasta_compute_length.xml]: It seems this file is mixing tabs with spaces, so here the indentation looks wrong.

File: tools/fasta_compute_length/fasta_compute_length.xml [hunk 2]
```diff
@@ -1,9 +1,13 @@
 <tool id="fasta_compute_length" name="Compute sequence length" version="1.0.0">
```
Comment (nsoranzo) [inline tools/fasta_compute_length/fasta_compute_length.xml]: Increase the version number.

**Comment by nsoranzo** [comment]
Where are the test data for this tool? :open_mouth:

**Comment by erasche** [comment]
@nsoranzo I get the impression it uses the built in Galaxy test data? Dunno. :/

**Comment by bgruening** [comment]
@erasche fixed @nsoranzo concers.

test-data is here: https://github.com/galaxyproject/galaxy-test-data
And we all probably have a cached version on our hardrive :)
The question is do we want to include it here, or is it enough to trigger planemo to download these test-data?

**Comment by peterjc** [comment]
A long overdue enhancement :)

**Comment by jmchilton** [comment]
Please don't add empty citations, hides a valid a problem :). Lets add:

```
<citations>
  <citation type="doi">10.1093/bioinformatics/btq281</citation>
</citations>
```

**Comment by bgruening** [comment]
@jmchilton Just to clarify, if there is no citation we can/should add an empty citations tag, right?

**Comment by jmchilton** [comment]
@bgruening Yes, but it should be fairly rare right? I'd also recommend a comment in these cases:

```<citations><!-- No citations because... --></citations>```

**Comment by bgruening** [comment]
Not sure, more or less all tools with a own python script don't have a citation. I was not aware of this publication, so sorry for the trouble. Can the devteam go through all the tools and add citations?

**Comment by jmchilton** [comment]
No trouble at all. I'd say most Python tools should cite something, the github repository they are in, an in preparation publication, etc....

As for citing all the tools, someday I guess :) Issue #3 (https://github.com/galaxyproject/tools-devteam/issues/3).

**Comment by nsoranzo** [comment]
@bgruening @erasche All tools in this repository (except for this, `dwt_var_perfeature` and `velvet`) have the test-data included, and I think this is how it was always supposed to be.

I'm going to open a separate PR for the other 2 tools, but it would be nice to fix `fasta_compute_length` here.

**Comment by nsoranzo** [comment]
:+1:

**Comment by bgruening** [comment]
Great thanks!

**Pull request merged**